### PR TITLE
trying to unify/simplify file system setup even further

### PR DESCRIPTION
**Pull request by ngc92**
now that the overall path handling is relatively simple, we can go one step further and make sure that every sub-directory that we are search for is added in all the base search paths that we want.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -76,48 +76,25 @@ void deinit()
...
-	fs.probeDir("scripts");
-	fs.probeDir("backgrounds");
-	fs.probeDir("rules");
-
-	/*
-	set read dir (local files next to application)
-	*/
+
+	// and add all reading locations
 	fs.addToSearchPath(baseSearchPath);
```
Comment (danielknobe) [inline src/main.cpp:88]: I'm not sure if I like this. We split a repeating path in two separate files (src/FileSystem.cpp - appendToSearchDirs)

File: src/main.cpp [hunk 2]
```diff
@@ -76,48 +76,25 @@ void deinit()
...
-	fs.addToSearchPath(baseSearchPath + "sounds.zip");
-	fs.addToSearchPath(baseSearchPath + "scripts.zip");
-	fs.addToSearchPath(baseSearchPath + "backgrounds.zip");
-	fs.addToSearchPath(baseSearchPath + "rules.zip");
-
-	/*
-	set read dir (unix-like when installed)
-	*/
 	#if BLOBBY_ON_DESKTOP && (!defined WIN32)
 		fs.addToSearchPath(BLOBBY_INSTALL_PREFIX  "/share/blobby");
```
Comment (danielknobe) [inline src/main.cpp:90]: same here.

Comment (ngc92) [reply src/main.cpp]: true. I think I've got a better approach now

**Comment by ngc92** [comment]
Instead of moving things inside the FileSystem class, I'm now just generating a local list of sub-dirs to add, and then loop over them. 
While looking into the BLOBBY_INSTALL_PREFIX thing, I've figured out that we actually can simplify this quite a lot, up to getting rid of the generated "config.h" file. We just provide the installation prefix as a `-D` definition from CMake. In a next step, we could even add a CMake option to make this configurable, might be handy for distro maintainers.

**Comment by danielknobe** [comment]
I like this patch. The path handling now fits on one screen page :+1: . Dropping the .in file is also a good idea.

**Pull request merged**